### DISCUSSION TITLE
Making an Assignment in Github Classroom Failing

### DISCUSSION THREAD

**Discussion by JohannesvKL** [community/community#129870] [GitHub Education]
### Github Classroom Assignment-Creation Failing

Hi Github Education team, 

when attempting to make an assignment for my class this week via Github classroom I received the error "Starter code repository must be a template so that a copy may be made in your organization". This wasn't a problem in the past weeks, and we've opted into creating student assignment repositories using forks., so I'm wondering if there's a fix besides turning the starter code repository into a template? 

Thanks in advance for your help.

**Comment by DavidMutchler** [comment]
My understanding of the latest changes (circa one week ago) is that in the new system (the one that will become mandatory July 1), the starter code MUST be a template, since the system now creates a copy of that template to use as the source for forking the student assignments.  That works for me now.  (It was broken a week ago.)

Edit 7/1:The problem recurred for me.  I believe that I have found the problem and a workaround.
**IT IS LIKELY TO AFFECT SOME OF YOU (not just summer users, but also new repos for the fall).**

The problem is that: (1) Github repository names are limited in length (to 100 characters, I think).  (2) The new system creates a new repository named by appending the CLASSROOM name and the STUDENT-REPO-PREFIX-NAME and the STARTER-REPO name.  Together, these sometimes exceeded the length limit and sometimes did not, hence my struggles above.

I have followed other's lead in choosing classroom names and student-repo-prefix names, and my starter-repo names are modest in length, so it is likely that some of you will encounter this problem.

The workaround is to shorten EITHER your classroom name OR your STUDENT-REPO-PREFIX-NAME OR the STARTER-CODE name when this occurs.  I chose to shorten the STUDENT-REPO-PREFIX-NAME since I want my STARTER-CODE names to remain fully descriptive.

**Comment by Rennish** [comment]
As a student what exactly should I do?

**Reply by DavidMutchler** [reply]
Nothing.  This is all regarding what INSTRUCTORS need to do.
